packages/openapi-fetch/test/common/response.test.ts

```diff
@@ -20,22 +20,38 @@ describe("response", () => {
       // 2. assert data is not undefined inside condition block
       if (result.data) {
         assertType<NonNullable<Resource[]>>(result.data);
-        assertType<undefined>(result.error);
+        // @ts-expect-error FIXME: This is a limitation within Typescript
```

Comment:
I love these tests, and this is a great start! But I fear that we’ll have to fix this before merging (it’ll be much more difficult to fix in a followup, when other work has built off the inference). I don’t really care if the type is `undefined` or `never`—that’s a detail—but TS should be able to infer the other property otherwise there could be issues!